src/mgr/MgrPyModule.cc

```diff
@@ -171,30 +171,45 @@ int MgrPyModule::load()
   }
 
   // Find the class
-  // TODO: let them call it what they want instead of just 'Module'
-  auto pClass = PyObject_GetAttrString(pModule, (const char*)"Module");
-  Py_DECREF(pModule);
-  if (pClass == nullptr) {
+  auto  mgr_module_type = PyObject_GetAttrString(pModule, (const char*)"MgrModule");
+
+  if (mgr_module_type == nullptr) {
     derr << "Class not found in module '" << module_name << "'" << dendl;
     derr << handle_pyerror() << dendl;
     return -EINVAL;
   }
 
-  
-  // Just using the module name as the handle, replace with a
-  // uuidish thing if needed
-  auto pyHandle = PyString_FromString(module_name.c_str());
-  auto pArgs = PyTuple_Pack(1, pyHandle);
-  pClassInstance = PyObject_CallObject(pClass, pArgs);
-  Py_DECREF(pClass);
-  Py_DECREF(pyHandle);
-  Py_DECREF(pArgs);
-  if (pClassInstance == nullptr) {
-    derr << "Failed to construct class in '" << module_name << "'" << dendl;
-    derr << handle_pyerror() << dendl;
-    return -EINVAL;
-  } else {
-    dout(1) << "Constructed class from module: " << module_name << dendl;
+  Py_DECREF(pModule);
```

Comment:
we cannot deref `pModule` yet. we are actually referencing it right after this line.